### ISSUE TITLE
Notation binder printing may capture constants

### ISSUE THREAD

**Issue by ggonthier** [coq/coq#9569]
The printing of notations does not take (named) binders into account when deciding whether to display global constants with a qualified name. In goals, this may lead to captures:
```
Goal exists I, I = Logic.I.
===> exists I : True, I = I.
```
Compare to:
```
Goal forall I, I = Logic.I.
===> forall I : True, I = Logic.I.
```
The issue only occurs at the head of the goal; in other contexts the bound variable is alpha converted to avoid any conflict
```
Goal (exists I, I = Logic.I) -> True.
===> (exists I0 : True, I0 = I) -> True.
```
Note that that strategy has its own drawbacks, since it may lead to unnecessary alpha conversion, when the global constant is not displayed as such, because of notation or implicits:
```
Goal (forall S, S = 1 * S) -> True.
===> (forall S0, S0 = 1 * S0) -> True.
```

**Comment by maximedenes** [comment]
@herbelin can I assign this issue to you?

**Comment by herbelin** [comment]
@ggonthier: regarding the first point, I made a fix at #12965.

Regarding the second point, I tend to agree with your taste and practical point. Don't know how to proceed though, to try to improve things a bit in a way which is relatively consensual.
